lib/bindings/python/Cargo.toml

```diff
@@ -30,8 +30,8 @@ repository = "https://github.com/ai-dynamo/dynamo.git"
 path = "rust/lib.rs"
 name = "_core"
 # "cdylib" is necessary to produce a shared library for Python to import from.
-crate-type = ["cdylib"]
-
+# "rlib" is necessary to support doctests.
+crate-type = ["cdylib", "rlib"]
```

Comment:
NOTE: this increases the size of the resultant artifact dir by about ~30%, and there currently aren't any doctests in `lib/bindings/python`, so I'm a bit conflicted about this change. However, without this, running `cargo test --doc` will fail because there are currently no targets to run on. If we add some doctests to this dynamo-py3 package, then this change makes more sense as the correct answer.

Another way to fix this would be to update the github action to skip running this step `cargo test --doc` altogether if `dir == lib/bindings/python`, but it felt cleaner to not have these conditionals in the github action. I could go either way on it.